### PR TITLE
fix: generation of Python imports

### LINKED ISSUE #974: Code generator does not create some imports
### Describe the bug

Declarations in the `safeds` package or any subpackage are always available. If explicit imports are missing, however, no imports are created in the Python code. This also leads to failures in the runner: `"name 'Table' is not defined"`.

### To Reproduce

Safe-DS:

```
package test

pipeline myPipeline {
    val table = Table.fromColumns([]);
}
```

Python:

```py
# Pipelines --------------------------------------------------------------------

def myPipeline():
    table = Table.from_columns([])
```

### Expected behavior

The following code should be created: 

```py
# Imports ----------------------------------------------------------------------

from safeds.data.tabular.containers import Table

# Pipelines --------------------------------------------------------------------

def myPipeline():
    table = Table.from_columns([])
```

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by lars-reimann**
Closes #974

### Summary of Changes

Fix the generation of Python imports. It now also works without explicit imports or if stubs are declared in the same module.

### CODE REVIEW

**Comment by lars-reimann** [comment]
@WinPlay02 It would be great if you could review this (not urgent). You know the code generator much better than I do.

File: packages/safe-ds-lang/src/language/generation/safe-ds-python-generator.ts [hunk 1]
```diff
@@ -1126,68 +1121,38 @@ export class SafeDsPythonGenerator {
...
+        if (importPath) {
+            const refText = reference.target.$refText;
             return {
-                importPath: `${this.getPythonModuleOrDefault(targetModule)}.${this.formatGeneratedFileName(
-                    this.getModuleFileBaseName(targetModule),
-                )}`,
-                declarationName: this.getPythonNameOrDefault(declaration),
+                importPath,
+                declarationName: this.getPythonNameOrDefault(target),
+                alias: refText === target.name ? undefined : refText,
```
[the remark this answers is not in the document]
Comment (lars-reimann) [reply packages/safe-ds-lang/src/language/generation/safe-ds-python-generator.ts:1151]: The idea here is that we want to detect whether the declaration is imported using an alias. This is the case if the `refText`, i.e. the text of the reference in the Safe-DS program, differs from the Safe-DS name of the referenced declaration.

Example:

```
package tests.generator.reference

@Impure([ImpurityReason.Other]) fun f(param: Any?)

@Pure
@PythonName("explain_model")
fun explainModel() -> result: Int

pipeline test {
    f(explainModel());
}
```

Output with `refText === target.name`:

```py
# Imports ----------------------------------------------------------------------

from tests.generator.reference import explain_model, f

# Pipelines --------------------------------------------------------------------

def test():
    f(explain_model())
```

Output with `refText === this.getPythonNameOrDefault(target)`:

```py
# Imports ----------------------------------------------------------------------

from tests.generator.reference import explain_model as explainModel, f

# Pipelines --------------------------------------------------------------------

def test():
    f(explainModel())

```

The latter adds an alias for all declarations where their `@PythonName` differs from their Safe-DS name. At first glance, both options seem correct. Are there any reasons to prefer one over the other?

[the remark this answers is not in the document]
Comment (WinPlay02) [reply packages/safe-ds-lang/src/language/generation/safe-ds-python-generator.ts:1152]: Then there should be no reason to use the python name here, thanks for the clarification.

**Review by WinPlay02 — APPROVED** [review]
Looks fine :+1:

**Pull request merged**